cli/src/commands/parse/pst.rs

```diff
@@ -0,0 +1,1275 @@
+    }
+
+    pub fn get_name(&self) -> Result<String> {
+        if self.parent_folder_path.is_root() {
+            return Ok(ROOT_FOLDER_NAME.to_string());
+        }
+
+        let size = self.get_name_size()?;
+
+        let mut utf8_string: Vec<c_char> = vec![0; size + 1];
+        let mut error: *mut LibcError = std::ptr::null_mut();
+        unsafe {
+            libpff_folder_get_utf8_name(self.inner, utf8_string.as_mut_ptr(), size, &mut error);
+        }
+        handle_error("Error getting folder name", error)?;
+
+        let string = unsafe { std::ffi::CStr::from_ptr(utf8_string.as_ptr()) };
+        Ok(string.to_string_lossy().to_string())
+    }
+
+    pub fn get_folder_path(&self) -> Result<FolderPath> {
+        let name = self.get_name()?;
+        Ok(self.parent_folder_path.sub_path(name))
+    }
+}
+
+struct SubMessagesIter {
+    idx: usize,
+    total_number_of_messages: usize,
+    folder: PstFolder,
+}
+
+impl SubMessagesIter {
+    fn new(folder: PstFolder) -> Result<Self> {
+        let total_number_of_messages = folder.get_number_of_sub_messages()?;
+
+        Ok(Self {
+            idx: 0,
+            total_number_of_messages,
+            folder,
+        })
+    }
+}
+
+impl Iterator for SubMessagesIter {
+    type Item = Result<PstMessage>;
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.idx < self.total_number_of_messages {
+            let item = self.folder.get_sub_message(self.idx);
+            self.idx += 1;
+            Some(item)
+        } else {
+            None
+        }
+    }
+}
+
+struct SubFoldersIter {
+    idx: usize,
+    total_number_of_folders: usize,
+    folder: PstFolder,
+}
+
+impl SubFoldersIter {
+    fn new(folder: PstFolder) -> Result<Self> {
+        let total_number_of_folders = folder.get_number_of_sub_folders()?;
+
+        Ok(Self {
+            idx: 0,
+            total_number_of_folders,
+            folder,
+        })
+    }
+}
+
+impl Iterator for SubFoldersIter {
+    type Item = Result<PstFolder>;
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.idx < self.total_number_of_folders {
+            let item = self.folder.get_sub_folder(self.idx);
+            self.idx += 1;
+            Some(item)
+        } else {
+            None
+        }
+    }
+}
+
+struct AttachmentsIter {
+    idx: usize,
+    total_num_attachments: usize,
+    message: PstMessage,
+}
+
+impl Iterator for AttachmentsIter {
+    type Item = Result<PstAttachment>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.idx < self.total_num_attachments {
+            let next_attachment = self.message.get_attachment(self.idx);
+            self.idx += 1;
+            Some(next_attachment)
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Clone)]
+struct PstMessage {
+    pub inner: *mut LibPffItem,
+    pub folder: FolderPath,
+}
+
+impl PstMessage {
+    // Useful for debugging
+    fn _get_message_id(&self) -> Result<String> {
```

Comment:
Generally Rust doesn't use a leading underscore for privacy - if you want to keep this private:
- Split `PstMessage` into it's own module, in which case this will be private an inacessible
- Call it `get_message_id_for_debug` or something
- Just call it `get_message_id`. Is there a reason you wouldn't want this used in other code?